### PR TITLE
Allow import/export of instance-level domain blocks/allows

### PR DESCRIPTION
**Pull request by Tak**
Rebase of https://github.com/mastodon/mastodon/pull/15664 per https://github.com/glitch-soc/mastodon/issues/1751

Adds an export button to the moderator federation page and an import block to the moderator add new domain block/allow page.
Blocked domains are exported along with metadata: severity level, whether media are rejected, whether reports are rejected, the public comment, and whether the block should be obfuscated.

On import, already-blocked domains will not be modified (e.g. a currently-silenced domain will not be upgraded to suspended).

### CODE REVIEW

**Comment by ClearlyClaire** [comment]
Thanks! I'm currently pretty busy as you can imagine, but I do appreciate the feature and will get it merged in some form or another when I do get the time to properly review it.

File: lib/paperclip/has_attached_file_extensions.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    def add_active_record_callbacks
+      name = @name
+      @klass.send(:after_save) { send(name).send(:save) }
+      @klass.send(:before_destroy) { send(name).send(:queue_all_for_delete) }
+      @klass.send(:after_destroy) { send(name).send(:flush_deletes) }
+    end
+  end
+end
+
+Paperclip::HasAttachedFile.prepend(Paperclip::HasAttachedFileExtensions)
```
Comment (ClearlyClaire) [inline lib/paperclip/has_attached_file_extensions.rb:15]: I'm not really comfortable with that change, why is it needed?

Comment (Tak) [reply lib/paperclip/has_attached_file_extensions.rb]: It was a hack to support `:on` with `after_commit` after the rails 6 upgrade, but it turns out the callback isn't needed anyway, so I've removed it along with the workaround

**Review by ClearlyClaire — APPROVED** [review]

**Comment by ClearlyClaire** [comment]
I'd eventually like to add a page between upload and the actual blocking is performed to sum up the blocks and let admins review them, but that can wait.

**Pull request merged**